compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -809,18 +807,33 @@ public void visit(BLangImportPackage importPkgNode) {
         this.env.scope.define(pkgAlias, symbol);
     }
 
-    public void initPredeclaredModules(Map<Name, BPackageSymbol> predeclaredModules, SymbolEnv env) {
+    public void initPredeclaredModules(Map<Name, BPackageSymbol> predeclaredModules,
+                                       List<BLangCompilationUnit> compUnits, SymbolEnv env) {
         SymbolEnv prevEnv = this.env;
         this.env = env;
-        for (Name alias : predeclaredModules.keySet()) {
-            ScopeEntry entry = this.env.scope.lookup(alias);
-            if (entry == NOT_FOUND_ENTRY) {
-                this.env.scope.define(alias, predeclaredModules.get(alias));
-            } else {
-                while (entry.next != NOT_FOUND_ENTRY) {
-                    entry = entry.next;
+        for (BLangCompilationUnit compUnit : compUnits) {
+            for (Name alias : predeclaredModules.keySet()) {
```

Comment:
@chiranSachintha, shall we check if we can switch the loops as discussed during the call?

Reply:
We can switch two loops but still, we need to check whether alias is defined or not after the first iteration. So do we need to switch these two loops?

Reply:
What about doing something like

```java
        for (Name alias : predeclaredModules.keySet()) {
            int index = 0;
            ScopeEntry entry = this.env.scope.lookup(alias);
            if (entry == NOT_FOUND_ENTRY) {
                this.env.scope.define(alias, dupPackageSymbolAndSetCompUnit(predeclaredModules.get(alias),
                            new Name(compUnits.get(index++).name)));
            }
            for (int i = index; i < compUnits.size(); i++) {
                boolean isUndefinedModule = true;
                ...... 
            } 
        }
```

Reply:
Changed.